### PR TITLE
Add support for serialising arrays of unboxed primitives

### PR DESCRIPTION
**Pull request by fenryka**
Arrays of objects are represented in the schema as <type> followed by [], so int[] for example. The problem here is that there is no mapping between that type and weather it is boxed or unboxed, so on deserialisation it will fail since the current assumption is that all types are boxed

To indicate an array of unboxed type we add the syntax <type>[p]

### CODE REVIEW

**Comment by mnesbit** [comment]
A little more explanation of why all the oddball cases.

File: core/src/main/kotlin/net/corda/core/serialization/amqp/ArraySerializer.kt [hunk 1]
```diff
@@ -7,12 +7,21 @@ import java.lang.reflect.Type
 /**
  * Serialization / deserialization of arrays.
```
Comment (mnesbit) [inline core/src/main/kotlin/net/corda/core/serialization/amqp/ArraySerializer.kt]: This is a java doc comment attached only to this method. A plain comment might be more sensible for this section break comment.

Comment (fenryka) [inline core/src/main/kotlin/net/corda/core/serialization/amqp/ArraySerializer.kt]: done!

**Comment by fenryka** [comment]
So commented in place why we're ignoring unboxed byte arrays (Since they coerce to the internal binary type by default)

**Pull request merged**